src/Microsoft.OData.Core/UriParser/Parsers/ODataPathParser.cs

```diff
@@ -1002,25 +1111,21 @@ private void TryBindKeySegmentIfNoResolvedParametersAndParenthesisValueExists(st
         /// <param name="previousSegment">The previous segment before the operation to be invoked.</param>
         /// <param name="identifier">The name of the segment</param>
         /// <param name="parenthesisExpression">The query portion</param>
+        /// <param name="bindingType">Type to which the operation is bound. It is an optional parameter that can be evalulated from the previous segment if it is null.</param>
         /// <returns>Whether or not the identifier referred to an action.</returns>

```

Comment:
In the case of the escape function, you actually know more than the binding type -- you know the actual specific function overload that you want to bind, parameters to pass, etc.  Rather than passing in (possibly null) values to this method, I was suggesting that you factor this method into two.  The first part of this method has the current signature (without the added bindingType), and it determines the binding type and specific function (i.e., everything down to 1143).  And then this method calls a new method (also called by TryBindEscapeFunction) that takes the previous segment, resolved function, and resolved parameters and does the bind.